parlai/mturk/tasks/acute_eval/run.py

```diff
@@ -0,0 +1,413 @@
+            except ValueError():
+                print("WARNING: couldn't remove task from worker's history")
+
+
+def get_onboarding_tasks(worker_id, tasks_per_hit):
+    """
+    Get the next onboarding task for this worker id.
+
+    If the worker has never done a task, shuffle the onboarding tasks for them. If
+    they've done all of the onboarding tasks or if there are no onboarding tasks, return
+    None
+    """
+    if len(onboarding_tasks) == 0:
+        return []
+    onboarding_tasks_todo = workers_to_onboarding_tasks_todo.get(worker_id)
+    if onboarding_tasks_todo is None:
+        onboarding_tasks_todo = list(onboarding_tasks.keys())
+        np.random.shuffle(onboarding_tasks_todo)
+        workers_to_onboarding_tasks_todo[worker_id] = onboarding_tasks_todo
+    if len(onboarding_tasks_todo) == 0:
+        return []
+    num_tasks_to_return = min(len(onboarding_tasks_todo), tasks_per_hit)
+    onboarding_tasks_chosen = onboarding_tasks_todo[:num_tasks_to_return]
+    workers_to_onboarding_tasks_todo[worker_id] = onboarding_tasks_todo[
+        num_tasks_to_return:
+    ]
+    return [onboarding_tasks[id] for id in onboarding_tasks_chosen]
+
+
+def check_and_update_worker_approval(mturk_manager, worker_id, threshold, save_data):
+    """
+    Soft block workers who fail onboarding tasks, keep track of their status.
+    """
+    all_task_data = save_data['worker_data'][worker_id]['task_data']
+    response_data = save_data['worker_data'][worker_id]['response']['task_data']
+    num_onboarding_tasks = 0
+    num_correct = 0
+    for i in range(len(all_task_data)):
+        is_onboarding = all_task_data[i]['pairing_dict'].get('is_onboarding', False)
+        if not is_onboarding:
+            continue
+        worker_response = response_data[i]['speakerChoice']
+        expected_response = all_task_data[i]['pairing_dict']['correct_answer']
+        num_onboarding_tasks += 1
+        if worker_response == expected_response:
+            num_correct += 1
+    if num_onboarding_tasks == 0:
+        if worker_id in onboarding_failed_workers:
+            return_task_data(worker_id, all_task_data)
+        return
+    if (num_correct / num_onboarding_tasks) >= threshold:
+        return
+    mturk_manager.soft_block_worker(worker_id)
+    onboarding_failed_workers.append(worker_id)
+
+
+def main(opt):
+    """
+    Handles setting up and running a ParlAI-MTurk task by instantiating an MTurk manager
+    and configuring it for the qa_data_collection task.
+    """
+    np.random.seed(opt['seed'])
+
+    # Set the task name to be the folder name
+    opt['task'] = os.path.basename(os.path.dirname(os.path.abspath(__file__)))
+    opt['task_description'] = {
+        'num_subtasks': opt['subtasks_per_hit'],
+        'question': opt['question'],
+    }
+    # append the contents of task_config.py to the configuration
+    opt['frontend_version'] = 1
+    try:
+        import sys
+
+        sys.path.append(opt['task_folder'])
```

Comment:
Explain this plz

Reply:
lol I'll just copy what I said to emily:

"We have a default task_config with the title, description, tags of the task to be shown on MTurk, but if the user writes one, I try to load it. I decided to go with having them write it in a .py file instead of .json (which would be easier to deal with) because I kept Jack's task_config.py template. There are a few lines on this in the README.
In other words, nothing is wrong if there's an exception, it just means the user didn't write a new task_config. But I should add a warning."

I guess this is super nonstandard though based on your reaction. Maybe I should switch to a .json task_config file.

Reply:
discussed offline, this is going into a command line arg